modules/reactor/src/meshgenerators/RevolveGenerator.C

```diff
@@ -0,0 +1,2340 @@
+#include "libmesh/cell_tet4.h"
+#include "libmesh/cell_tet10.h"
+#include "libmesh/cell_tet14.h"
+#include "libmesh/cell_hex8.h"
+#include "libmesh/cell_hex20.h"
+#include "libmesh/cell_hex27.h"
+#include "libmesh/face_tri3.h"
+#include "libmesh/face_tri7.h"
+#include "libmesh/face_quad4.h"
+#include "libmesh/face_quad9.h"
+#include "libmesh/point.h"
+
+// C++ includes
+#include <cmath>
+
+registerMooseObject("ReactorApp", RevolveGenerator);
+
+InputParameters
+RevolveGenerator::validParams()
+{
+  InputParameters params = PolygonMeshGeneratorBase::validParams();
+  params.addClassDescription("This RevolveGenerator object is designed to revolve a 1D mesh into "
+                             "2D, or a 2D mesh into 3D based on an axis.");
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh to revolve");
+
+  params.addRequiredParam<Point>("axis_point", "A point on the axis of revolution");
+
+  params.addRequiredParam<Point>("axis_direction", "The direction of the axis of revolution");
+
+  params.addRangeCheckedParam<std::vector<Real>>("revolving_angles",
+                                                 "revolving_angles<=360.0 & revolving_angles>0.0",
+                                                 "The angles to revolve around the axis");
+
+  params.addParam<std::vector<std::vector<subdomain_id_type>>>(
+      "subdomain_swaps",
+      {},
+      "For each row, every two entries are interpreted as a pair of "
+      "'from' and 'to' to remap the subdomains for that azimuthal section");
+
+  params.addParam<std::vector<std::vector<boundary_id_type>>>(
+      "boundary_swaps",
+      {},
+      "For each row, every two entries are interpreted as a pair of "
+      "'from' and 'to' to remap the boundaries for that elevation");
+
+  params.addParam<std::vector<std::string>>(
+      "elem_integer_names_to_swap",
+      {},
+      "Array of element extra integer names that need to be swapped during revolving.");
+
+  params.addParam<std::vector<std::vector<std::vector<dof_id_type>>>>(
+      "elem_integers_swaps",
+      {},
+      "For each row, every two entries are interpreted as a pair of 'from' and 'to' to remap the "
+      "element extra integer for that elevation. If multiple element extra integers need to be "
+      "swapped, the enties are stacked based on the order provided in "
+      "'elem_integer_names_to_swap' to form the third dimension.");
+
+  params.addParam<boundary_id_type>(
+      "start_boundary", "The boundary ID to set on the starting boundary for partial revolving.");
+
+  params.addParam<boundary_id_type>(
+      "end_boundary", "The boundary ID to set on the ending boundary for partial revolving.");
+
+  params.addParam<bool>(
+      "clockwise", true, "Revolve clockwise around the axis or not (i.e., counterclockwise)");
+
+  params.addRequiredParam<std::vector<unsigned int>>(
+      "nums_azimuthal_intervals", "The numbers of azimuthal intervals to revolve around the axis");
```

Comment:
is it more like the azimuthal interval discretization for each azimuthal section?